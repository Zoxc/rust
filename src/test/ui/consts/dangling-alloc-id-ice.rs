// run-pass
// https://github.com/rust-lang/rust/issues/55223

union Foo<'a> {
    //~^ WARN union is never used
    y: &'a (),
    long_live_the_unit: &'static (),
}

const FOO: &() = {
    //~^ WARN constant item is never used
    let y = ();
    unsafe { Foo { y: &y }.long_live_the_unit }
};

fn main() {}
