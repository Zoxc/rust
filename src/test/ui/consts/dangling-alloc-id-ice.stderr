warning: union is never used: `Foo`
  --> $DIR/dangling-alloc-id-ice.rs:4:7
   |
LL | union Foo<'a> {
   |       ^^^
   |
   = note: `#[warn(dead_code)]` on by default

warning: constant item is never used: `FOO`
  --> $DIR/dangling-alloc-id-ice.rs:10:1
   |
LL | / const FOO: &() = {
LL | |
LL | |     let y = ();
LL | |     unsafe { Foo { y: &y }.long_live_the_unit }
LL | | };
   | |__^

